src/index.js

```diff
@@ -111,23 +109,29 @@ const copyToNewDir = (fileName, newImage, cb) => {
     });
 };
 
+const buildFileName = (name, test) => {
+  const fileNameParts = [];
+  if (name) {
+    fileNameParts.push(name);
+  }
+  let currentTest = test;
+  while (currentTest) {
+    fileNameParts.push(currentTest.title);
+    currentTest = currentTest.parent;
+  }
+  return fileNameParts.reverse().join(' -- ');
+};
+
 /**
  * Takes a screenshot and, if available, matches it against the screenshot
  * from the previous test run. Assertion will fail if the diff is larger than
  * the specified threshold
- * @param  {String} name
  * @param  {Object} options
  */
-function matchScreenshot(
```

Comment:
Moved `name` into options to make usage like this cleaner: https://github.com/FreeWillPBC/freewill-api-v2/blob/0f73add0b429b74b16630be787ce6e22d7a0ca61/src/web/cypress/integration/qcd.spec.js#L85-L90